maggy/core/experiment_driver/OptimizationDriver.py

```diff
@@ -32,46 +37,40 @@ class OptimizationDriver(Driver):
         "gp": bayes.GP,
         "none": SingleRun,
         "faulty_none": None,
-        "gridsearch": GridSearch,
     }
 
-    def __init__(
-        self,
-        num_trials,
-        optimizer,
-        searchspace,
-        direction,
-        es_policy,
-        es_interval,
-        es_min,
-        name,
-        description,
-        num_executors,
-        hb_interval,
-        log_dir,
-    ):
-        # num_trials default 1
-        # direction default 'max'
-        super().__init__(
-            name, description, direction, num_executors, hb_interval, log_dir
-        )
-        # CONTEXT-SPECIFIC EXPERIMENT SETUP
-        self.exp_type = "optimization"
-        self.num_trials = num_trials
-        self.searchspace = self._init_searchspace(searchspace)
-        self.controller = self._init_controller(optimizer, self.searchspace)
+    def __init__(self, config, num_executors, log_dir):
+        super().__init__(config, num_executors, log_dir)
+        self._final_store = []
+        self._trial_store = {}
+        self.experiment_done = False
+        self.maggy_log = ""
+
+        if isinstance(
+            config, AblationConfig
+        ):  # Interrupt init, dirty fix for deviating config.
+            return
```

Comment:
why not raise a TypeException if it's the wrong config object?

Reply:
It's not the wrong type. AblationDriver inherits from OptimizationDriver. I am not too happy about this, but AblationDriver implements basically the exact same message queue callbacks as OptimizationDriver. So in order to avoid copying all those callbacks into AblationDriver I inherited them. Some of them could surely be pruned, but I was reluctant to dive deep into the ablation communication. Since the AblationConfig does not have a searchspace, super().__init__() needs to be interrupted before it reaches config.searchspace. Hence the "dirty fix" comment.
To avoid this, we can either copy all the callbacks into AblationDriver and base it on Driver again, or leave it like this. In the end, I think it would be best if Sina implements only the callbacks he really needs in AblationDriver and changes the inheritance back.

Reply:
yes okay, can you open an issue on the repo and add what you wrote here as description, and assign it to sina.